troposphere/__init__.py

```diff
@@ -38,8 +38,11 @@ def __init__(self, title, template=None, **kwargs):
                            'Metadata', 'UpdatePolicy',
                            'Condition', 'CreationPolicy']
 
-        # unset/None is also legal
-        if title and not valid_names.match(title):
+        # some objects allow more characters than others
+        if title and 'validate_title' in dir(self):
```

Comment:
Does it maybe make more sense to do away with this logic and instead move the `valid_names.match` into a default `validate_title` method?  It'd make this a bit cleaner.  This would become:

```
if title:
    self.validate_title()
```